Code/Tools/FBuild/FBuildTest/Tests/TestIf.cpp

```diff
@@ -42,11 +44,16 @@ REGISTER_TESTS_BEGIN( TestIf )
     REGISTER_TEST( IfFunctionBool )
     REGISTER_TEST( IfFunctionInt )
     REGISTER_TEST( IfFunctionStringCompare )
+    REGISTER_TEST( IfFunctionBracket )
     REGISTER_TEST( UsageError_ExtraTokensAfterExpression )
     REGISTER_TEST( UsageError_UnsupportedTypeForIn )
     REGISTER_TEST( UsageError_UnsupportedOperation )
 REGISTER_TESTS_END
 
+#define TEST_EXP_TRUE( vars, exp )        TEST_ASSERT( ParseFromString( true, vars "\nIf ( " exp " )\n{\nPrint('Success')\n}", "Success") );
+#define TEST_EXP_FALSE( vars, exp )       TEST_ASSERT( ParseFromString( true, vars "\nIf ( " exp " )\n{\nPrint('Failure')\n}", nullptr, "Failure") );
+#define TEST_EXP_FAIL( vars, exp, error ) TEST_ASSERT( ParseFromString( false, vars "\nIf ( " exp " )\n{\n}", error ) );
```

Comment:
These macros seems to somewhat or entirely duplicate those in FBuildTest.h
```C++
#define TEST_PARSE_OK( ... )        TEST_ASSERT( ParseFromString( true, __VA_ARGS__ ) )
#define TEST_PARSE_FAIL( ... )      TEST_ASSERT( ParseFromString( false, __VA_ARGS__ ) )
```

Reply:
They are inspired by `TEST_PARSE_OK/FAIL` but they are dedicated to function `If`.
By using `TEST_EXP_<...>`, the test cases focus to the expressions within the header of `If`, with less boilerplate code.

For `TEST_PARSE_FALSE`, it even relies on the updated version of `ParseFromString(...., unexpectedMessage)`, meaning we cannot achieve the same goal via `TEST_PARSE_OK/FAIL`.

Reply:
I am going to click the "reslove" button for I didn't receive further feedback.